src/AmplNLWriter.jl

```diff
@@ -23,31 +23,58 @@ immutable AmplNLSolver <: AbstractMathProgSolver
     filename::String
 end
 
-osl = isdir(Pkg.dir("CoinOptServices"))
-ipt = isdir(Pkg.dir("Ipopt"))
-
-if osl; import CoinOptServices; end
-if ipt; import Ipopt; end
-
 function AmplNLSolver(solver_command::String,
                       options::Vector{String}=String[];
                       filename::String="")
     AmplNLSolver(solver_command, options, filename)
 end
 
 function BonminNLSolver(options=String[]; filename::String="")
-    osl || error("CoinOptServices not installed. Please run\n",
-                 "Pkg.add(\"CoinOptServices\")")
-    AmplNLSolver(CoinOptServices.bonmin, options; filename=filename)
+    error("""
+        BonminNLSolver is no longer available by default through AmplNLWriter.
```

Comment:
This is a big breaking change and needs corresponding updates to the README and JuMP's solver table.

Reply:
Yeah I'm not planning on merging this any time soon. Just trying it out. Yay/Nay on whether this is a good idea?

Reply:
Go for it. This dependency is very flaky.